client/src/components/EmailDialog.js

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+
+const useStyles = makeStyles((theme) => ({
+    confirmBtn: {
+        paddingTop: 10,
+        paddingBottom: 10,
+        marginBottom: 5,
+        color: theme.palette.primary.main,
+        borderColor: theme.palette.primary.main,
+        alignItems: 'center',
+        '&:hover': {
+            color: '#fff',
+            backgroundColor: theme.palette.primary.main
+        },
+        width: '48.5%',
+        display: 'inline-block!important'
+    },
+}));
+
+export default function EmailDialog(props) {
+    
+    const classes = useStyles();
+
+    const [open, setOpen] = useState(false);
+    const [email, setEmail] = useState('');
+
+    const handleClickOpen = () => {
+        setOpen(true);
+    };
+
+    const handleClose = () => {
+        setOpen(false);
+    };
+
+    const handleEmailChange = (e) => {
+        setEmail(e.target.value)
+    }
+
+    const handleSubmit = () => {
+        let status;
+        const dateString = new Date(props.selectedDay).toISOString().substring(0, 10);
+        const dateTimeString = new Date(dateString + ' ' + props.selectedTime).toISOString();
+        const timezone = new Date().getTimezoneOffset(); 
+        // offset for now, can change to "America/vancouver" format depends on needs
+        
+        fetch('/appointment', {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json"
+            },
+            credentials: "include",
+            body: JSON.stringify({
+                dateTime: dateTimeString,
+                timezone: timezone,
+                email: email
```

Comment:
- dateTimeString format: `2021-03-28T11:11:53.859Z`
- timezone format: `420` in minute, need to /60 for hours. But might need to change to other formats depends on usage.

Reply:
Needs to add the person's name. 

Also need a way to figure out which type of event was booked. 

Also, did we ever finish the event link functionality? i.e. clicking on url to get to the calendar page?

Reply:
the copy link button on the cards in dashboard is working, but we need to figure out a way to make the url work for open a unique calendar.